src/tqdm_publisher/_demos/_parallel_bars/_server.py

```diff
@@ -207,12 +215,18 @@ def update_queue(request_id, id, n, total, **kwargs):
 
 
 def run_parallel_bar_demo() -> None:
-    """Asynchronously start the servers"""
-    asyncio.run(start_server(PORT))
+    """Asynchronously start the servers."""
+    asyncio.run(start_server(port=PORT))
+
+
+def _run_parallel_bars_demo(port: str, host: str):
+    URL = f"http://{HOST}:{PORT}"
 
+    request_id = uuid.uuid4()
+    run_parallel_processes(all_task_times=TASK_TIMES, request_id=request_id, url=URL)
 
-if __name__ == "__main__":
 
+if __name__ == "main":
```

Comment:
I'm assuming you like the `__name__ == "main"` ability for debugging purposes?

In lieu of just running the respective functions (with custom inputs) from an interactive python console?

I'd otherwise like to minimize the number of CLI ways of calling these demos to simplify what we have to support

Reply:
I also plan on reducing the amount of globally communicated static variables in a follow-up

Anything global gets pickled, sent, and exposed to spawned processes when multiprocessing and that can lead to unexpected behaviors (not saying that's causing a problem, or is essential to how this works, just mentioning its usually better to minimize the chance)

Reply:
We can remove the `__name__ == "main"` and just debug like this in separate files. Originally, I was spawning each file as a child process, so this was actually useful for demo execution.

Otherwise sounds good! Feel free to modify those globals